### PR TITLE
Fix Travis CI build and clean up

### LINKED ISSUE #369: Fix Travis CI
```
0.30s$ sudo apt-get install cmake libXxf86vm-dev libxinerama-dev libxcursor-dev

Reading package lists... Done

Building dependency tree       

Reading state information... Done

E: Unable to locate package libXxf86vm-dev

The command "sudo apt-get install cmake libXxf86vm-dev libxinerama-dev libxcursor-dev" failed and exited with 100 during .
```

### PR DESCRIPTION
**Pull request by vignesh-sankaran**
Resolves #369.

An overview of the changes made:

The error `E: Unable to locate package libXxf86vm-dev` occurs because there is a typo in the package name, the correct name of the package is `libxxf86vm-dev` (lower case x). In any case, this package is installed by default in Trusty so I removed it.

The build was running into problems with a CMake build which stated that `The RandR library and headers were not found`. After looking [here](https://github.com/openMVG/openMVG/issues/85), this was resolved by installing the `xorg-dev` and  `libglu1-mesa-dev`, but since the latter is already installed in Trusty I just added the former.

CMake is now at 3.2.2 in Trusty and installed by default so I removed this package from the list that are installed.

In summary, the dependencies added are:
- `xorg-dev`

Dependencies removed:
- `cmake`
- `libxxf86vm-dev`

I changed the previous way of installing dependencies to use the [TravisCI apt addon](https://docs.travis-ci.com/user/installing-dependencies/#Installing-Packages-with-the-APT-Addon). It is in beta, but is more human-readable.

I also removed the environment variables to `/usr/local/lib` and the secure key, and also removed the commands post script related to Rust CI. The latter is a service that has been replaced by docs.rs.

### CODE REVIEW

**Comment by vignesh-sankaran** [comment]
I have also removed the notes regarding Travis CI in the README.

**Comment by bvssvni** [comment]
Nice! Merging.
